src/librbd/AioImageRequest.cc

```diff
@@ -724,10 +732,140 @@ void AioImageFlush<I>::send_image_cache_request() {
   image_ctx.image_cache->aio_flush(req_comp);
 }
 
+template <typename I>
+void AioImageWriteSame<I>::assemble_extent(const ObjectExtent &object_extent,
```

Comment:
Sorry -- I was getting confused w/ ObjectExtent::buffer_extents. You will actually want to loop over the buffer extents since that is the mapping back to your original writesame buffer.